### PR TITLE
change behavior of mode='a' to be like python open (issue #1144)

### PR DESCRIPTION
**Pull request by jswhit**

### CODE REVIEW

**Comment by zerothi** [comment]
I would love this. :)
Thanks!

File: src/netCDF4/_netCDF4.pyx [hunk 1]
```diff
@@ -2246,7 +2255,7 @@ strings.
                     ierr = nc_open(path, NC_NOWRITE | NC_SHARE, &grpid)
                 else:
                     ierr = nc_open(path, NC_NOWRITE, &grpid)
-        elif mode == 'r+' or mode == 'a':
+        elif mode == 'r+' or mode == 'a' and os.path.exists(filename):
```
Comment (zerothi) [inline src/netCDF4/_netCDF4.pyx:2258]: `and` and `or` does not have the same precedence, this is read as `mode == r+ or (mode == a and os.path.exists(filename)`, didn't you mean: `mode in ('r+', 'a') and os.path.exists(filename)`? Same problem further down.

Comment (zerothi) [inline src/netCDF4/_netCDF4.pyx:2258]: In any case, putting parenthesis might be more clear about intent

Comment (jswhit2) [reply src/netCDF4/_netCDF4.pyx:2258]: good catch - will fix

**Pull request merged**